libcontainer/container_linux.go

```diff
@@ -195,12 +191,23 @@ func (c *linuxContainer) Run(process *Process) error {
 	if err != nil {
 		return err
 	}
-	isInit := status == Stopped
-	if err := c.start(process, isInit); err != nil {
+	if err := c.start(process, status == Stopped); err != nil {
 		return err
 	}
-	if isInit {
-		return process.ops.signal(InitContinueSignal)
+	if status == Stopped {
+		return c.Exec()
+	}
+	return nil
+}
+
+func (c *linuxContainer) Exec() error {
+	f, err := os.OpenFile(filepath.Join(c.root, execFifoFilename), os.O_RDONLY, 0)
+	if err != nil {
+		return newSystemErrorWithCause(err, "open exec fifo for reading")
+	}
+	defer f.Close()
+	if _, err := ioutil.ReadAll(f); err != nil {
```

Comment:
Why we can't use a pipe like in user namespace writing trick from @mrunalp in go syscall package? It seems like FIFO require strange assumptions about go scheduler behavior.